src/OrbitQt/orbitmainwindow.cpp

```diff
@@ -1302,17 +1227,15 @@ void OrbitMainWindow::OnProcessListUpdated(
   UpdateCaptureStateDependentWidgets();
 }
 
-// TODO(170468590): [ui beta] When out of ui beta, this can return TargetConfiguration (without
-// std::optional)
-std::optional<orbit_qt::TargetConfiguration> OrbitMainWindow::ClearTargetConfiguration() {
-  using StadiaTarget = orbit_qt::StadiaTarget;
-  if (target_configuration_.has_value() &&
-      std::holds_alternative<StadiaTarget>(target_configuration_.value())) {
-    std::get<StadiaTarget>(target_configuration_.value())
-        .GetProcessManager()
-        ->SetProcessListUpdateListener(nullptr);
-  }
-  std::optional<orbit_qt::TargetConfiguration> result = std::move(target_configuration_);
-  target_configuration_ = std::nullopt;
-  return result;
+void OrbitMainWindow::ShutdownTarget(const orbit_qt::StadiaTarget& target) {
+  target.GetProcessManager()->SetProcessListUpdateListener(nullptr);
+}
+
+void OrbitMainWindow::ShutdownTarget(const orbit_qt::LocalTarget& target) {
+  target.GetProcessManager()->SetProcessListUpdateListener(nullptr);
+}
```

Comment:
Have you considered adding a member function to the target-classes called `Shutdown` or something similar?

Reply:
Yes, but what I just realized, the actual problem is that this is poorly named. 

That the function does is only remove the things from the target that belong to the mainwindow. Not everything of the target will be "shutdown". For example the grpc connection will not be cancelled, since it can be reused again in the startup window.

There are multiple options
1. I could remove the `ShutdownTarget` function and then the `ClearTargetConfiguration` function would look like this:
```
orbit_qt::TargetConfiguration OrbitMainWindow::ClearTargetConfiguration() {
  if (std::holds_alternative<orbit_qt::StadiaTarget>(target_configuration_)) {
    std::get<orbit_qt::StadiaTarget>(target_configuration_)
        .GetProcessManager()
        ->SetProcessListUpdateListener(nullptr);
  } else if (std::holds_alternative<orbit_qt::LocalTarget>(target_configuration_)) {
    std::get<orbit_qt::LocalTarget>(target_configuration_)
        .GetProcessManager()
        ->SetProcessListUpdateListener(nullptr);
  }
  return std::move(target_configuration_);
}
```

2. Rename the `ShutdownTarget` function to one of those (this also needs to be defined for `FileTarget` which is somewhat awkward)
a. `ClearProcessListCallback`
b. `ClearProcessListCallbackOnTarget`
c. `RemoveMainWindowArtifactsFromTarget`
d. `ShutdownProcessListUpdate`

I think I prefer 2 a, but I would like to have your opinions

Reply:
To me what is actually happening (clearing the process list callback) is an implementation detail. That's why I don't like 2a, 2b and 2d so much.

Next I would consider which part of the code is actually responsible for this to happen. Is the main window or is it the target class?

Since this code is specific to the main window, I would argue it's the main windows's responsibility. So the code should also live there.

So I would vote for 1.

Reply:
Okay. I change it to option 1. I will rebase at the same time and then merge afterwards